failing doctest
```
sage -t --long --random-seed=286735480429121101562228604801325644303 sage/dynamics/arithmetic_dynamics/dynamical_semigroup.py
**********************************************************************
File "sage/dynamics/arithmetic_dynamics/dynamical_semigroup.py", line 933, in sage.dynamics.arithmetic_dynamics.dynamical_semigroup.DynamicalSemigroup.__mul__
Failed example:
    f*g
Expected:
    Traceback (most recent call last):
    ...
    TypeError: can only multiply dynamical semigroups with other dynamical semigroups of the same type
Got:
    <BLANKLINE>
    Traceback (most recent call last):
      File "/sage/src/sage/doctest/forker.py", line 709, in _run
        self.compile_and_execute(example, compiler, test.globs)
      File "/sage/src/sage/doctest/forker.py", line 1144, in compile_and_execute
        exec(compiled, globs)
      File "<doctest sage.dynamics.arithmetic_dynamics.dynamical_semigroup.DynamicalSemigroup.__mul__[46]>", line 1, in <module>
        f*g
        ~^~
      File "/sage/src/sage/dynamics/arithmetic_dynamics/dynamical_semigroup.py", line 967, in __mul__
        raise ValueError("left dynamical semigroup's domain must equal right dynamical semigroup's codomain")
    ValueError: left dynamical semigroup's domain must equal right dynamical semigroup's codomain
**********************************************************************
1 item had failures:
   1 of  62 in sage.dynamics.arithmetic_dynamics.dynamical_semigroup.DynamicalSemigroup.__mul__
    [398 tests, 1 failure, 5.31 s]
```

--- a/src/sage/dynamics/arithmetic_dynamics/dynamical_semigroup.py
+++ b/src/sage/dynamics/arithmetic_dynamics/dynamical_semigroup.py
@@ -961,7 +961,7 @@ class DynamicalSemigroup(Parent, metaclass=InheritComparisonClasscallMetaclass):
             ...
             ValueError: left dynamical semigroup's domain must equal right dynamical semigroup's codomain
         """
-        if not isinstance(other_dynamical_semigroup, DynamicalSemigroup):
+        if type(self) is not type(other_dynamical_semigroup):
             raise TypeError("can only multiply dynamical semigroups with other dynamical semigroups of the same type")
         if self.domain() != other_dynamical_semigroup.codomain():
             raise ValueError("left dynamical semigroup's domain must equal right dynamical semigroup's codomain")

